frontend/src/shared/layout/navigation/navigation.reducer.ts

```diff
@@ -51,10 +51,10 @@ const navigationSlice = createSlice({
                 state.adminItems = undefined;
             }
         },
-        setItemsForProjectName (state, action: PayloadAction<string | undefined>) {
-            const projectName = action.payload;
+        setItemsForProjectName (state, action: PayloadAction<any | undefined>) {
```

Comment:
Can we define a type for this so we're not relying on `any`? Looks like `{projectName: string, enabledServices: IEnabledServices}` would probably work.

Reply:
Do we need the app config value here? The appConfig is in the store, which this reducer should have access to. Why do we need to pass this value in? Additionally if we are directly accessing the current config value, then we don't need to update the input parameters on all of the dispatch calls.

Reply:
This is my first experience working with Thunk but I was under the impression that wasn't how it worked. 

> Sometimes a slice reducer really needs to read a value that isn't in its own slice of state. A possible workaround to that is to dispatch a thunk, extract the needed values from state, and then dispatch a plain action containing the additional info.

Reply:
@estohlmann As far as I can tell you're not using thunk just a vanilla react reducer which makes sense since you're not making an async call.

@LightSeekerSC I don't believe he will have access to the app config in this reducer since `state` is available but will only be the navigation state/slice and not the top level application state which he would need.

Reply:
Okay, this makes me not like Redux. @dustins is right though and as the comment @estohlmann quoted, Thunk does allow this. The navigation reducer doesn't use Thunk, but the app config does. I tested this, and it *should* work.

You can import `useAppDispatch` in `configuration-reducer.ts` and then within the `listConfigurations` function, if you change it so that it doesn't immediately return the axios response, but saves it, then you can dispatch the `data[0]` element's `EnabledServices` of the response to update EnabledServices within the navigation element instead of doing it in multiple other places. This is still kind of gross, but it consolidates the logic and I believe that it will work